tests/Fixer/ArrayNotation/MultilineLongArrayFixerTest.php

```diff
@@ -0,0 +1,434 @@
+            <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => ['baz' => ['foo']],];
+            INPUT,
+        ];
+
+        yield 'Single line array with brackets inside of a string' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => 'foo is [baz]',
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => 'foo is [baz]',];
+            INPUT,
+        ];
+
+        yield 'Single line array with arrow function' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => fn($i) => 'baz'
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => fn($i) => 'baz'];
+            INPUT,
+        ];
+
+        yield 'Single line array with anonymous function' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => function( $x ,$y) { return $x + $y; }
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => function( $x ,$y) { return $x + $y; }];
+            INPUT,
+        ];
+
+        yield 'Internal short arrays inside long one.' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [
+                'foo' => ['short' => $array,],
+                'bar' => ['other' => 'short','array' => 3,],
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = [
+                'foo' => [
+                    'short' => $array,
+                ],
+                'bar' => [
+                    'other' => 'short',
+                    'array' => 3,
+                ],
+            ];
+            INPUT,
+            ['max_length' => 40],
+        ];
+
+        yield 'don\'t change anonymous class implements list but change array inside' => [
+            <<<'EXPECTED'
+            <?php
+            $x = [
+            1,
+            "2",
+            "c" => new class implements Foo, Bar { const FOO = [
+            "x",
+            "y"
+            ]; },
+            $y
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $x = [1,  "2","c" => new class implements Foo, Bar { const FOO = ["x","y"]; },$y ];
+            INPUT,
+        ];
+
+        yield 'don\'t change anonymous class implements list and don\'t change small array inside.' => [
+            <<<'EXPECTED'
+            <?php
+            $x = [
+            1,
+            "2",
+            "c" => new class implements Foo, Bar { const FOO = ["x","y"]; },
+            $y
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $x = [1, "2", "c" => new class implements Foo, Bar { const FOO = ["x","y"]; },$y ];
+            INPUT,
+            ['max_length' => 15],
+        ];
+    }
```

Comment:
Can we have a test for input array like:
```php
$letters = ['a', /* @todo: add some other letters one day */ 'z'];
```
and `max_length` of `20`?

Reply:
Good Idea.
I've added the test but I'm not sure about what we want it to return.

Here is the current output :
```php
<?php
$letters = [
"a",
/* @todo: add some other letters one day */ "z"
];
```

I think one of these would be more appropriate :
```php
<?php
$letters = [
"a", /* @todo: add some other letters one day */
"z"
];

$letters = [
"a",
/* @todo: add some other letters one day */
"z"
];
```
In any case, it could break users' habits when adding comments to arrays so another solution would be to skip comments entierly (right now they are included in max_length).
Any suggestions ?

Reply:
There are also comments in multi-line arrays, which are intended to be put in other places :
```php
<?php
$letters = [
"a", // This is the letter a
"z", /** This is not the letter a */
];

// becomes
<?php
$letters = [
"a",
// This is the letter a
"z",
/** This is not the letter a */
];

// Acceptable solutions could be 
<?php
$letters = [
// This is the letter a
"a",
/** This is not the letter a */
"z",
];

<?php
$letters = [
"a", // This is the letter a
"z", /** This is not the letter a */
];
```

Reply:
Here's how #6387 handles comments (I think it's equally unsatisfying) :
```php
<?php $a5 = [1,2,3,4,     /* 0 */5,/* 1 */ /* 2 */  6,/* 6+ */7,];

// becomes 
<?php $a5 = [
1,
2,
3,
4,     /* 0 */
5,/* 1 */ /* 2 */
6,/* 6+ */
7,
];
```

I believe there are 2 reasons someone would put comments inside an array :
- To explain values (see previous comment)
- - It mostly applies to multi-line arrays so I don't think we should change the shape of these arrays.
- To temporarily remove an element (see code above)
- - It would be possible to treat these as regular elements and apply formatting rules.

In any case, we would need to replace `//` with `/** */` for single-line arrays.

Since I don't see a reliable way of differenciating these two cases, I think the best solution would be to skip arrays with comments altogether, which is the change I just made to the PR.